routes/auth.js

```diff
@@ -46,24 +25,35 @@ router.post('/login', async (req, res) => {
   }
 });
 
-router.get('/logout', async (req, res) => {
+router.get('/logout', authHelpers.loginRequired, async (req, res) => {
   req.session.destroy();
   return res.redirect('/');
-  // res.render('customError', {
-  //   title: 'Logout',
-  //   message: 'You have been succesfully logged out of the application.',
-  // });
 });
 
-router.get('/signup', async (req, res) => {
+router.get('/signup', authHelpers.isLoggedIn, async (req, res) => {
+  let roles = await Role.find(
+    { name: { $in: ['Donor', 'Recipient'] } },
+    '_id name'
+  );
   res.render('auth/signup', {
     title: 'Sign Up',
+    roles: roles,
   });
 });
 
-router.post('/create', async (req, res) => {
-  console.log('inside user post');
-  return res.redirect('/');
+router.post('/register', authHelpers.isLoggedIn, async (req, res) => {
+  try {
+    let userInfo = req.body;
+    console.log(userInfo);
+    userData.validateUserInfo(userInfo);
+    let user = await userData.create(userInfo);
+    if (user) {
+      req.session.user = user;
+    }
+    res.redirect('/');
+  } catch (error) {
+    res.status(400).json({ error: error });
```

Comment:
422 is more relevant here.

```
The HyperText Transfer Protocol (HTTP) 422 Unprocessable Entity response status code 
indicates that the server understands the content type of the request entity, and the 
syntax of the request entity is correct, but it was unable to process the 
contained instructions.
```

Reply:
@abhinav098 I will read more on this and will update later accordingly.